### ISSUE TITLE
Ability to configure port forwarding in kubernetes environment

### ISSUE THREAD

**Issue by vzhukovs** [containers/podman-desktop#7078]
### Is your enhancement related to a problem? Please describe

Due to analysing previous feedbacks. There was a feature request that is missing in Podman Desktop. The feature called Port Forwarding. It should represent `kubectl port-forward` cli command.

### Describe the solution you'd like

Need to create the backend implementation, which will serve port forwards and UI dedicated page to control the port forwarding state.

Main things that need to be implemented:

- [ ] ability to store forward configurations (add/list/remove) https://github.com/containers/podman-desktop/pull/7379
- [ ] ability to start and stop saved port forward https://github.com/containers/podman-desktop/pull/7379
- [ ] dedicated page on the UI that shows the list of forwards
- [ ] dedicated form on the UI that allows to register new port forward based on the input type of service (pod/service/deployment)

**Comment by benoitf** [comment]
@vzhukovs looks like the description of the issue is empty and not matching our templates

**Comment by vzhukovs** [comment]
@benoitf I've updated the issue description to met the current requirements

**Comment by cdrage** [comment]
With regards to UI, we'll most likely have to add it as well as a separate section similar to how we list services, deployments, etc.

This is usually standard on other UI's as well I've seen.

**Comment by vzhukovs** [comment]
> With regards to UI, we'll most likely have to add it as well as a separate section similar to how we list services, deployments, etc.
> 
> This is usually standard on other UI's as well I've seen.

Yep, it definitely should be a dedicated page on the UI to manage the forwards. The main idea is to have the list of forwards with start/stop and edit/delete buttons on each item. In the top there can be a create button that will open a form to create a new forward with necessary input fields. Each item in the list can represent a single port mapping or a bulk one. A bit later will try to make some mockups for it to have agree on the final UI version.

**Comment by cdrage** [comment]
@vzhukovs I also want to ask if we will be supporting any controller / service / pod?

Looking through the `kubectl port-forward --help`, it supports doing port forwarding on a deployment, pod and service.

I'm assuming we'll be adding it as an action to all of them?

**Comment by vzhukovs** [comment]
> I also want to ask if we will be supporting any controller / service / pod?

@cdrage the backend code supports forwarding ports from pods, deployments, services (simple numeric ports and named ports), the only one limitation at this moment is that everything forwards to localhost, passing different host address will be implemented in the future PRs. It depends, what workload kind we will pass to the backend side from the UI.

The main idea is to allow user to select where ports will be taken to forward. Some kind of selection input in create forward configuration dialog.
